### PR TITLE
Bump version requirementslib and more vendor clean up

### PR DESCRIPTION
**Pull request by oz123**
As we now bumped requirementslib, we could now also bump vistir to 0.7.4.
This in turn forces us to make some tiny changes, and allows us to finally drop yaspin.

### CODE REVIEW

**Comment by oz123** [comment]
@matteius all tests pass, besides Python3.7, but this is unrelated to the changes I did.

File: pipenv/utils/resolver.py [hunk 1]
```diff
@@ -63,6 +63,16 @@
 err = rich.console.Console(stderr=True)
 
 
+class FakePipTempDirdirectory:
```
Comment (matteius) [inline pipenv/utils/resolver.py:66]: I think having the name Fake in a class that is used in the actual production code is confusing, though I see why you added this.  Also it has `Dirdirectory`  Can we name it  `PipenvTempDirectoryWrapper` or something similar instead?

Comment (oz123) [reply pipenv/utils/resolver.py:66]: Can we use  `QuasiPipTempDirectory`  instead? Wrapper is misleading because we aren't wrapping anything.

Comment (matteius) [reply pipenv/utils/resolver.py:66]: Well we are wrapping the directory name 🤷   I am not sure it is Quasi either?

Comment (matteius) [reply pipenv/utils/resolver.py:66]: Ah its fine -- better than Fake.

**Review by matteius — APPROVED** [review]

**Pull request merged**